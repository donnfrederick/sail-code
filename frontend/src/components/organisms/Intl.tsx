import React, {
  PropsWithChildren,
  Dispatch,
  SetStateAction,
  useContext,
  useState
} from 'react'
import { IntlProvider } from 'react-intl'
import { messages, browserLocale, Locale } from 'utils/intl'

export type LocaleState = {
  locale: Locale
}

const initialState: LocaleState = {
  locale: browserLocale
}

const IntlStateContext = React.createContext<LocaleState>(initialState)

const SetLocaleStateContext = React.createContext<
  Dispatch<SetStateAction<LocaleState>>
>(() => {
  // empty block
})

export function useLocaleState() {
  return useContext(IntlStateContext)
}
export function useSetLocaleState() {
  return useContext(SetLocaleStateContext)
}

export function LocaleStateProvider(props: {
  initialState?: LocaleState
  children: React.ReactNode
}) {
  const [state, setState] = useState<LocaleState>(
    props.initialState || initialState
  )
  return (
    <IntlStateContext.Provider value={state}>
      <SetLocaleStateContext.Provider value={setState}>
        {props.children}
      </SetLocaleStateContext.Provider>
    </IntlStateContext.Provider>
  )
}

export const Intl = (props: PropsWithChildren<{}>) => {
  const localeState = useLocaleState()

  return (
    <LocaleStateProvider>
      <IntlProvider
        messages={messages[localeState.locale]}
        locale={localeState.locale}
        defaultLocale="en"
      >
        {props.children}
      </IntlProvider>
    </LocaleStateProvider>
  )
}
