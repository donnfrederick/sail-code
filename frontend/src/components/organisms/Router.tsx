import * as Sentry from '@sentry/browser'
import ErrorPage from 'components/pages/Error'
import { createBrowserHistory } from 'history'
import * as React from 'react'
import { Provider, Store } from 'react-redux'
import { ConnectedRouter } from 'react-router-redux'
import { compose, lifecycle, withState } from 'recompose'
import createStore from 'store/createStore'
import resetViewport from 'utils/resetViewport'
import userAgent from 'utils/userAgent'

declare const gtag: any

export const history = createBrowserHistory()
export const store = createStore(history) as Store<any>

history.listen(() => {
  window.scrollTo(0, 0)
  gtag('config', 'UA-105776977-2', { page_path: history.location.pathname })
  userAgent()
  resetViewport()
})

interface Props {
  children: JSX.Element
  errors: Error | null
  setErrors: (state: any) => any
}

const isProd = process.env.NODE_ENV === 'production'

const sentry = compose(
  withState('errors', 'setErrors', null),
  isProd
    ? lifecycle<Props, {}, {}>({
        componentDidCatch(error, errorInfo) {
          this.props.setErrors(error)
          Sentry.withScope(scope => {
            Object.keys(errorInfo).forEach(key => {
              scope.setExtra(key, errorInfo[key])
            })
            Sentry.captureException(error)
          })
        }
      })
    : lifecycle<Props, {}, {}>({
        componentDidCatch(error) {
          this.props.setErrors(error)
        }
      })
)

export default sentry((props: Props) => {
  const { children, errors } = props

  return (
    <Provider store={store}>
      <ConnectedRouter history={history}>
        {errors ? <ErrorPage /> : children}
      </ConnectedRouter>
    </Provider>
  )
})
