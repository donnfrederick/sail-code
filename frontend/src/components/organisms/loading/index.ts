import Loading from 'components/organisms/loading/Loading'
import { connect } from 'react-redux'
import { State as RootState } from 'reducers'

// tslint:disable-next-line
interface OuterProps {}

interface Props {
  isFetchingConversation: boolean
  isFetchingLocations: boolean
  isFetchingNotification: boolean
  isFetchingOrganizations: boolean
  isFetchingStudentsMe: boolean
  isFetchingTeachersMe: boolean
}

const mapStateToProps = (state: RootState, props: OuterProps): Props => {
  return {
    isFetchingConversation: state.rootReducer.conversation.fetching > 0,
    isFetchingLocations: state.rootReducer.locations.isFetching,
    isFetchingNotification: state.rootReducer.notification.isFetching,
    isFetchingOrganizations: state.rootReducer.organizations.isFetching,
    isFetchingStudentsMe: state.rootReducer.students.isFetching,
    isFetchingTeachersMe: state.rootReducer.teachers.isFetching
  }
}

const enhancer: any = connect(
  mapStateToProps,
  {}
)

export default enhancer(Loading)
