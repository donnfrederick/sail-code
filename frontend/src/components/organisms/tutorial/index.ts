import Tutorial from 'components/organisms/tutorial/Tutorial'
import { connect } from 'react-redux'
import { State as RootState } from 'reducers'
import * as TutorialActions from 'reducers/tutorial'

// tslint:disable-next-line
interface OuterProps {}

interface Props {
  shouldShow: boolean
  showConfirmation: boolean
}

const mapStateToProps = (state: RootState, props: OuterProps): Props => {
  return {
    shouldShow: state.rootReducer.tutorial.shouldShow,
    showConfirmation: state.rootReducer.teachers.showConfirmation
  }
}

const enhancer: any = connect(
  mapStateToProps,
  {
    hide: TutorialActions.hide,
    show: TutorialActions.show
  }
)

export default enhancer(Tutorial)
