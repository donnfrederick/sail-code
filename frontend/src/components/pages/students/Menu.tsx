import StudentsMenu from 'components/templates/students/Menu'
import * as React from 'react'

export default () => {
  return <StudentsMenu />
}
