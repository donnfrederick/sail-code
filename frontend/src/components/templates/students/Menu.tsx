import Footer from 'components/molecules/students/Footer'
import MenuList, { Item } from 'components/molecules/students/MenuList'
import Header from 'components/organisms/students/header'
import Signout from 'components/organisms/students/signout'
import clearInfo from 'hocs/clearInfo'
import * as React from 'react'
import styled from 'styled-components'
import isWebView from 'utils/isWebView'
import resolvePath from 'utils/resolvePath'
import { FormattedMessage } from 'react-intl'

export default clearInfo(() => {
  return (
    <Container>
      <MenuList items={accountItems} subHeading="Account" marginBottom={48} />
      <FormattedMessage id="menu.support" defaultMessage="Support">
        {chunks => (
          <MenuList
            items={supportItems}
            subHeading={chunks ? chunks[0] : 'Support'}
            marginBottom={48}
          />
        )}
      </FormattedMessage>
      <FormattedMessage id="menu.legal" defaultMessage="Legal">
        {chunks => (
          <MenuList
            items={legalItems}
            subHeading={chunks ? chunks[0] : 'Legal'}
            marginBottom={48}
          />
        )}
      </FormattedMessage>
      <FormattedMessage id="menu." defaultMessage="About">
        {chunks => (
          <MenuList
            items={aboutItems}
            subHeading={chunks ? chunks[0] : 'About'}
            marginBottom={48}
          />
        )}
      </FormattedMessage>
      <Signout />
      <FormattedMessage id="templates.Menu.Header" defaultMessage="Menu">
        {chunks => (
          <Header
            text={chunks}
            hasBackButton={true}
            returnPath={resolvePath.page('students', 'mypage')}
          />
        )}
      </FormattedMessage>

      <Footer />
    </Container>
  )
})

const accountItems: Item[] = [
  {
    intlid: 'menu.my_page',
    link: resolvePath.page('students', 'profile'),
    text: 'My Page'
  },
  {
    intlid: 'menu.edit_profile',
    link: resolvePath.page('students', 'profile/edit/info'),
    text: 'Edit Profile'
  },
  {
    intlid: 'menu.change_password',
    link: resolvePath.page('students', 'profile/edit/password'),
    text: 'Change Password'
  }
]

const supportItems: Item[] = [
  {
    intlid: 'menu.help_center',
    link: resolvePath.page('students', 'faq'),
    text: 'Help Center'
  },
  {
    intlid: 'menu.conversation_history',
    link: resolvePath.page('students', 'history'),
    text: 'Conversation History'
  },
  {
    intlid: 'menu.favorites_users',
    link: resolvePath.page('students', 'favorites'),
    text: 'Favorite Japanese'
  },
  {
    intlid: 'menu.blocked_accounts',
    link: resolvePath.page('students', 'blocked'),
    text: 'Blocked Accounts'
  }
]

const legalItems: Item[] = [
  {
    // link: resolvePath.page('students', 'privacy'),
    intlid: 'menu.privacy_policy',
    link: 'https://www.helte.jp/privacy-policy-en/',
    text: 'Privacy Policy'
  },
  {
    // link: resolvePath.page('students', 'terms'),
    intlid: 'menu.terms',
    link: 'https://www.helte.jp/terms-en/',
    text: 'Terms'
  }
]

if (isWebView()) {
  legalItems.push({
    link: resolvePath.page('students', 'license'),
    text: 'License',
    intlid: 'menu.license'
  })
}

const aboutItems: Item[] = [
  {
    link: resolvePath.page('students', 'about'),
    text: 'Operating Company',
    intlid: 'menu.operating_company'
  }
]

const Container = styled.div`
  position: relative;
  width: 100%;
  padding: 152px 0;
`
