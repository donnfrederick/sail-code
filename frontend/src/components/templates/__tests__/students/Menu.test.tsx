import Router from 'components/organisms/Router'
import Menu from 'components/templates/students/Menu'
import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { Intl } from 'components/organisms/Intl'

test('render <Menu />', () => {
  const tree = renderer
    .create(
      <Intl>
        <Router>
          <Menu />
        </Router>
      </Intl>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
