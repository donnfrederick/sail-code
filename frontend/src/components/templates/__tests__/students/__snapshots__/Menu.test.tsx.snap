// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <Menu /> 1`] = `
<div
  className="sc-eHgmQL iFlQFf"
>
  <div
    className="sc-gqjmRU gggsyL"
  >
    <div
      className="sc-VigVT jpTuN"
    >
      Account
    </div>
    <div
      className="sc-jTzLTM gLnvy"
    >
      <div
        className="sc-htoDjs bUzAjn"
      >
        <a
          className="sc-gZMcBi dTGJGz"
          href="/students/profile"
          onClick={[Function]}
        >
          <div
            className="sc-dnqmqq hkiqYL"
          >
            My Page
          </div>
          <img
            className="sc-iwsKbI fIyQcL"
            src="/assets/img/students/profile/arrow@3x.png"
          />
        </a>
      </div>
      <div
        className="sc-htoDjs bUzAjn"
      >
        <a
          className="sc-gZMcBi dTGJGz"
          href="/students/profile/edit/info"
          onClick={[Function]}
        >
          <div
            className="sc-dnqmqq hkiqYL"
          >
            Edit Profile
          </div>
          <img
            className="sc-iwsKbI fIyQcL"
            src="/assets/img/students/profile/arrow@3x.png"
          />
        </a>
      </div>
      <div
        className="sc-htoDjs bUzAjn"
      >
        <a
          className="sc-gZMcBi dTGJGz"
          href="/students/profile/edit/password"
          onClick={[Function]}
        >
          <div
            className="sc-dnqmqq hkiqYL"
          >
            Change Password
          </div>
          <img
            className="sc-iwsKbI fIyQcL"
            src="/assets/img/students/profile/arrow@3x.png"
          />
        </a>
      </div>
    </div>
  </div>
  <div
    className="sc-gqjmRU gggsyL"
  >
    <div
      className="sc-VigVT jpTuN"
    >
      Support
    </div>
    <div
      className="sc-jTzLTM gLnvy"
    >
      <div
        className="sc-htoDjs bUzAjn"
      >
        <a
          className="sc-gZMcBi dTGJGz"
          href="/students/faq"
          onClick={[Function]}
        >
          <div
            className="sc-dnqmqq hkiqYL"
          >
            Help Center
          </div>
          <img
            className="sc-iwsKbI fIyQcL"
            src="/assets/img/students/profile/arrow@3x.png"
          />
        </a>
      </div>
      <div
        className="sc-htoDjs bUzAjn"
      >
        <a
          className="sc-gZMcBi dTGJGz"
          href="/students/history"
          onClick={[Function]}
        >
          <div
            className="sc-dnqmqq hkiqYL"
          >
            Conversation History
          </div>
          <img
            className="sc-iwsKbI fIyQcL"
            src="/assets/img/students/profile/arrow@3x.png"
          />
        </a>
      </div>
      <div
        className="sc-htoDjs bUzAjn"
      >
        <a
          className="sc-gZMcBi dTGJGz"
          href="/students/favorites"
          onClick={[Function]}
        >
          <div
            className="sc-dnqmqq hkiqYL"
          >
            Favorited Users
          </div>
          <img
            className="sc-iwsKbI fIyQcL"
            src="/assets/img/students/profile/arrow@3x.png"
          />
        </a>
      </div>
      <div
        className="sc-htoDjs bUzAjn"
      >
        <a
          className="sc-gZMcBi dTGJGz"
          href="/students/blocked"
          onClick={[Function]}
        >
          <div
            className="sc-dnqmqq hkiqYL"
          >
            Blocked Accounts
          </div>
          <img
            className="sc-iwsKbI fIyQcL"
            src="/assets/img/students/profile/arrow@3x.png"
          />
        </a>
      </div>
    </div>
  </div>
  <div
    className="sc-gqjmRU gggsyL"
  >
    <div
      className="sc-VigVT jpTuN"
    >
      Legal
    </div>
    <div
      className="sc-jTzLTM gLnvy"
    >
      <div
        className="sc-htoDjs bUzAjn"
      >
        <a
          href="https://www.helte.jp/privacy-policy-en/"
        >
          <div
            className="sc-dnqmqq hkiqYL"
          >
            Privacy Policy
          </div>
        </a>
      </div>
      <div
        className="sc-htoDjs bUzAjn"
      >
        <a
          href="https://www.helte.jp/terms-en/"
        >
          <div
            className="sc-dnqmqq hkiqYL"
          >
            Terms
          </div>
        </a>
      </div>
    </div>
  </div>
  <div
    className="sc-gqjmRU gggsyL"
  >
    <div
      className="sc-VigVT jpTuN"
    >
      About
    </div>
    <div
      className="sc-jTzLTM gLnvy"
    >
      <div
        className="sc-htoDjs bUzAjn"
      >
        <a
          className="sc-gZMcBi dTGJGz"
          href="/students/about"
          onClick={[Function]}
        >
          <div
            className="sc-dnqmqq hkiqYL"
          >
            Operating Company
          </div>
          <img
            className="sc-iwsKbI fIyQcL"
            src="/assets/img/students/profile/arrow@3x.png"
          />
        </a>
      </div>
    </div>
  </div>
  <div
    className="sc-hSdWYo eXlnsO"
    onClick={[Function]}
  >
    Sign Out
  </div>
  <header
    className="sc-kpOJdX eZErHz"
  >
    <div
      className="sc-dxgOiQ lkITdM"
    >
      Menu
    </div>
    <a
      className="sc-jKJlTe iZdiKE"
      href="/students/mypage"
      onClick={[Function]}
      theme="blue"
    />
  </header>
  <div
    className="sc-gzVnrw bbOeQD"
  >
    <a
      className="sc-EHOje gEifQf"
      href="/students/mypage"
      icon="/assets/img/students/footer/home@3x.png"
      onClick={[Function]}
    />
    <a
      className="sc-EHOje bIkBAA"
      href="/students/notifications"
      icon="/assets/img/students/footer/notification@3x.png"
      onClick={[Function]}
    />
    <a
      className="sc-EHOje exXkhS"
      href="/students/mypage"
      icon="/assets/img/students/footer/cart.png"
      onClick={[Function]}
    />
    <a
      className="sc-bZQynM iAxnUN"
      href="/students/reservations/new"
      icon="/assets/img/students/footer/calendar@3x.png"
      onClick={[Function]}
    />
  </div>
</div>
`;
